exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/ParquetGroupScan.java

```diff
@@ -791,6 +799,43 @@ public FileGroupScan clone(FileSelection selection) throws IOException {
   }
 
   @Override
+  public GroupScan applyLimit(long maxRecords) {
```

Comment:
We do see small Parquet files (typically Hive generated via partitioning on date), although probably bigger than the typical LIMIT value.  I am ok with not doing this optimization for now.